nixos/tests/installer.nix

```diff
@@ -232,25 +233,29 @@ let
 
             # The test cannot access the network, so any packages we
             # need must be included in the VM.
-            system.extraDependencies = with pkgs;
-              [ sudo
-                libxml2.bin
-                libxslt.bin
-                docbook5
-                docbook5_xsl
-                unionfs-fuse
-                ntp
-                nixos-artwork.wallpapers.gnome-dark
-                perlPackages.XMLLibXML
-                perlPackages.ListCompare
-                xorg.lndir
-
-                # add curl so that rather than seeing the test attempt to download
-                # curl's tarball, we see what it's trying to download
-                curl
-              ]
-              ++ optional (bootLoader == "grub" && grubVersion == 1) pkgs.grub
-              ++ optionals (bootLoader == "grub" && grubVersion == 2) [ pkgs.grub2 pkgs.grub2_efi ];
+            system.extraDependencies = let
+              # This is because the EFI version of grub overrides the base
+              # version of GRUB 2, so we can't just use pkgs.grub2_efi here.
+              grub2 = pkgs.grub2_light.override { zfsSupport = grubUseZfs; };
+              grub2Efi = grub2.override { efiSupport = grubUseEfi; };
+            in with pkgs; [
+              sudo
+              libxml2.bin
+              libxslt.bin
+              docbook5
+              docbook5_xsl
+              unionfs-fuse
+              ntp
+              nixos-artwork.wallpapers.gnome-dark
+              perlPackages.XMLLibXML
+              perlPackages.ListCompare
+              xorg.lndir
+
+              # add curl so that rather than seeing the test attempt to download
+              # curl's tarball, we see what it's trying to download
+              curl
+            ] ++ optional (bootLoader == "grub" && grubVersion == 1) pkgs.grub
+              ++ optionals (bootLoader == "grub" && grubVersion == 2) [ grub2 grub2Efi ];
```

Comment:
It feels a bit like much of this diff is moving code just to make the formatting slightly nicer. That can make it much more difficult to read the diff, for a minor improvement.

Reply:
There is `git diff -w` which should make it way easier to follow.

Reply:
(also GitHub has an option `Hide whitespace changes`, which does something similar)